java/test/jmri/jmrit/operations/trains/XmlTest.java

```diff
@@ -26,6 +28,68 @@
  * @version $Revision$
  */
 public class XmlTest extends OperationsTestCase {
+    
+    public void testFilePathNames() {
+        // test the build report path name
+        Assert.assertEquals("buildstatus", TrainManagerXml.BUILD_STATUS);
+        Assert.assertEquals(OperationsXml.getFileLocation() +
+                "operations" +
+                File.separator +
+                "JUnitTest" +
+                File.separator +
+                "buildstatus" +
```

Comment:
Completely agree with you.  More interested in checking that the path names aren’t broken as they have been since 4.0.1.  And the tested directory names need to stay constants as they are referenced by the operation’s help.

 

Dan

 

From: Randall Wood [mailto:notifications@github.com] 
Sent: Wednesday, December 02, 2015 8:06 PM
To: JMRI/JMRI <JMRI@noreply.github.com>
Cc: Daniel Boudreau <daboudreau@hotmail.com>
Subject: Re: [JMRI] Add tests to check operation file pathnames (#302)

 

In java/test/jmri/jmrit/operations/trains/XmlTest.java <https://github.com/JMRI/JMRI/pull/302#discussion_r46501116> :

> @@ -26,6 +28,68 @@
>   * @version $Revision$
>   */
>  public class XmlTest extends OperationsTestCase {
> +    
> +    public void testFilePathNames() {
> +        // test the build report path name
> +        Assert.assertEquals("buildstatus", TrainManagerXml.BUILD_STATUS);
> +        Assert.assertEquals(OperationsXml.getFileLocation() +
> +                "operations" +
> +                File.separator +
> +                "JUnitTest" +
> +                File.separator +
> +                "buildstatus" +

These strings can be constants. Given that a constant is unchanging, there is no advantage to not using the constant.

—
Reply to this email directly or view it on GitHub <https://github.com/JMRI/JMRI/pull/302/files#r46501116> .  <https://github.com/notifications/beacon/ANw2p8IPyH29OuOqEZsWcMkuvJjZc0gxks5pL40TgaJpZM4GtoW8.gif>